### PR TITLE
Add missing plugin messaging channel

### PR DESCRIPTION
**Pull request by olijeffers0n**
Adds `MessageRaw` with component example

### CODE REVIEW

**Comment by Nacioszeczek** [comment]
Perhaps using a JSON component serializer would be way cleaner than the JSON itself?

**Comment by MiniDigger** [comment]
yeah, really dont wanna encourage hard coding json strings in any way

**Comment by olijeffers0n** [comment]
Yeah we discussed this last night. Updated it to use adventure

**Review by MiniDigger — APPROVED** [review]

File: docs/paper/dev/api/plugin-messaging.md [hunk 1]
```diff
@@ -194,3 +195,29 @@ This is not a recommended way to ban players due to the fact that there may not
...
+public class MyPlugin extends JavaPlugin {
+    
+    @Override
+    public void onEnable() {
+        this.getServer().getMessenger().registerOutgoingPluginChannel(this, "BungeeCord");
+        
+        Player player = ...;
+        ByteArrayDataOutput out = ByteStreams.newDataOutput();
+        out.writeUTF("MessageRaw");
+        out.writeUTF("ALL");
```
Comment (NoahvdAa) [inline docs/paper/dev/api/plugin-messaging.md:214]: The text above the snippet make it sound like this is only going to one player, when this is being broadcasted to all players on the proxy. This should be clarified to mention that this should be the name of the target player or ALL.

File: docs/paper/dev/api/plugin-messaging.md [hunk 2]
```diff
@@ -194,3 +195,29 @@ This is not a recommended way to ban players due to the fact that there may not
 but it is an example of how this can be used.
 
 :::
+
+#### MessageRaw
+
+The `MessageRaw` message type is used to send a raw chat component to a player. This is useful for sending clickable messages to players,
+or messages with hover events.
```
Comment (NoahvdAa) [inline docs/paper/dev/api/plugin-messaging.md:202]: Maybe also mention sending to players that aren't on the same backend server as an extra use case.

**Review by NoahvdAa — CHANGES_REQUESTED** [review]

**Comment by olijeffers0n** [comment]
Thanks, Addressed comments

**Comment by NoahvdAa** [comment]
Now that I think of it, "This is useful for sending clickable messages to players," sounds weird considering we (and Spigot) have native support for chat components. I feel like this might be a leftover in the bungee docs from a time where this native integration didn't exist yet. Maybe we're better off removing this entire part? I don't really see the purpose for this channel anymore apart from the aforementioned option of using it to send messages to players on a different server.

**Comment by olijeffers0n** [comment]
Good point. I think this is more just bad wording...  but yeah the fact that it is a clickable component is not really significant anymore.

**Review by NoahvdAa — APPROVED** [review]

**Pull request merged**